src/pages/tasks/NewTaskDetailsPage.js

```diff
@@ -63,7 +77,7 @@ const NewTaskPage = (props) => {
         >
             <HeaderWithCloseButton
                 title={props.translate('newTaskPage.assignTask')}
-                onCloseButtonPress={() => Navigation.dismissModal()}
+                onCloseButtonPress={() => TaskUtils.dismissModalAndClearOutTaskInfo()}
                 shouldShowBackButton
                 onBackButtonPress={() => Navigation.goBack()}
```

Comment:
@akinwale On this page alone, kindly clear out the task `TaskUtils.dismissModalAndClearOutTaskInfo()` since back action here closes the modal and has to reset the state.

Reply:
@abdulrahuman5196 Done.